### PR TITLE
feat: add create offices component

### PR DESCRIPTION
**Pull request by Rishabh106**
# Description
Add routing to Create Office button in Office directory, implemented the API and designed the UI.

## Related issues and discussion
#147 

## Screenshots, if any
create-office1
create-office2

## Checklist
Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] If you have multiple commits please combine them into one commit by squashing them.

- [x] Read and understood the contribution guidelines at `web-app/.github/CONTRIBUTING.md`.

### CODE REVIEW

File: src/app/organization/offices/create-offices/create-offices.component.html [hunk 1]
```diff
@@ -0,0 +1,48 @@
+<div class="container">
```
Comment (abhaychawla) [inline src/app/organization/offices/create-offices/create-offices.component.html]: Make sure the container width is small as the form is short as per the task description. (See Accounting -> Closing Entries -> Create as an example)

File: src/app/organization/offices/create-offices/create-offices.component.html [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  <mat-card>
+    <form [formGroup]="newOfficeForm" (ngSubmit)="submit()">
```
Comment (abhaychawla) [inline src/app/organization/offices/create-offices/create-offices.component.html]:
```suggestion
    <form [formGroup]="officeForm" (ngSubmit)="submit()">
```
Naming convention should be consistent with the other components of the web app. Make the required changes.

File: src/app/organization/offices/create-offices/create-offices.component.html [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+          <mat-form-field fxFlex="100%">
+            <mat-label>Opened On</mat-label>
+            <input matInput [min]="minDate" [max]="maxDate" formControlName="openingDate" required [matDatepicker]="openedOnDatePicker">
+            <mat-datepicker-toggle matSuffix [for]="openedOnDatePicker"></mat-datepicker-toggle>
+            <mat-datepicker #openedOnDatePicker></mat-datepicker>
+            <mat-error *ngIf="newOfficeForm.controls.openingDate.hasError('required')">
+              Opening Date is <strong>required</strong>
+            </mat-error>
+          </mat-form-field>
+          <mat-form-field fxFlex="100%">
```
Comment (abhaychawla) [inline src/app/organization/offices/create-offices/create-offices.component.html]: Ensure proper indentation/spacing in the entire file. Have a look at the other component for reference.

File: src/app/organization/offices/create-offices/create-offices.component.html [hunk 4]
```diff
@@ -0,0 +1,48 @@
+<div class="container">
+  <mat-card>
+    <form [formGroup]="newOfficeForm" (ngSubmit)="submit()">
+      <mat-card-content>
+        <div fxLayout="column" fxLayoutAlign="center">
+          <mat-form-field fxFlex="100%">
+            <mat-label>Office</mat-label>
+            <input required matInput formControlName="name" />
```
Comment (abhaychawla) [inline src/app/organization/offices/create-offices/create-offices.component.html]:
```suggestion
            <input required matInput formControlName="name">
```
Remove the `/` from all inputs.

File: src/app/organization/offices/create-offices/create-offices.component.ts [hunk 5]
```diff
@@ -0,0 +1,51 @@
+import { Component, OnInit } from '@angular/core';
+import { FormGroup, FormBuilder, Validators } from '@angular/forms';
+import { OrganizationService } from '../../organization.service';
+import * as _moment from 'moment';
```
Comment (abhaychawla) [inline src/app/organization/offices/create-offices/create-offices.component.ts]: Don't use `moment`, follow what is being done as of now within the other components as per this guide https://github.com/openMF/web-app/wiki/Date-and-Language

File: src/app/organization/offices/create-offices/create-offices.component.ts [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+import { Router, ActivatedRoute } from '@angular/router';
+
+@Component({
+  selector: 'mifosx-create-office',
+  templateUrl: './create-offices.component.html',
+  styleUrls: ['./create-offices.component.scss']
+})
+
```
Comment (abhaychawla) [inline src/app/organization/offices/create-offices/create-offices.component.ts]: Add comments, remove extra space as in other components.

File: src/app/organization/offices/create-offices/create-offices.component.ts [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+export class CreateOfficesComponent implements OnInit {
+  newOfficeForm: FormGroup;
+  officeData: any;
+  minDate = new Date(1990, 0, 1);
+  maxDate = new Date();
+  constructor(private formBuilder: FormBuilder, private organizationService: OrganizationService, private router: Router,
```
Comment (abhaychawla) [inline src/app/organization/offices/create-offices/create-offices.component.ts]: Properly indent wherever necessary.

File: src/app/organization/organization-routing.module.ts [hunk 8]
```diff
@@ -88,6 +104,22 @@ const routes: Routes = [
               }
             }
           ]
+        },
+        {
+          path: 'tellers',
```
Comment (abhaychawla) [inline src/app/organization/organization-routing.module.ts]: Resolve merge conflicts in the PR.

File: src/app/organization/organization.service.spec.ts [hunk 9]
```diff
@@ -1,12 +1,15 @@
-import { TestBed } from '@angular/core/testing';
```
Comment (abhaychawla) [inline src/app/organization/organization.service.spec.ts]: Don't make changes to this file. Its not a part of this task.

File: src/app/organization/organization.service.ts [hunk 10]
```diff
@@ -32,6 +32,10 @@ export class OrganizationService {
     return this.http.get('/offices');
   }
```
Comment (abhaychawla) [inline src/app/organization/organization.service.ts]: Add necessary comments.

File: src/app/organization/offices/create-offices/create-offices.component.html (hunk 1 above)
Comment (Rishabh106) [reply src/app/organization/offices/create-offices/create-offices.component.html]: is 50% width good for the form?

**Pull request closed without merging**